agent/taskresource/cgroupstatus.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package taskresource
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+)
+
+// CgroupStatus defines resource statuses for cgroups
+type CgroupStatus ResourceStatus
+
+const (
+	// CgroupStatusNone is the zero state of a task resource
+	CgroupStatusNone CgroupStatus = iota
+	// CgroupCreated represents a task resource which has been created
+	CgroupCreated
+	// CgroupCleaned represents a task resource which has been cleaned up
+	CgroupCleaned
+)
+
+var cgroupStatusMap = map[string]CgroupStatus{
+	"NONE":    CgroupStatusNone,
+	"CREATED": CgroupCreated,
+	"CLEANED": CgroupCleaned,
+}
+
+// StatusString returns a human readable string representation of this object
+func (cs CgroupStatus) String() string {
+	for k, v := range cgroupStatusMap {
+		if v == cs {
+			return k
+		}
+	}
+	return "NONE"
+}
+
+// TaskStatus maps the resource status to the corresponding task status.
+func (cs CgroupStatus) TaskStatus() api.TaskStatus {
```

Comment:
I'm not sure if we need this method at all. Is there a similar method in `VolumeResource` too? We have this for `ContainerStatus` because task's status is somewhat tied to container's status. But, that shouldn't be the case here.

cc @yunhee-l

Reply:
We have this for `VolumeResource`. Thinking about this, it should probably be `ContainerStatus`, since container status can be tied to the resource states. What do you think?

Reply:
I was quite a bit confused initially on whether the resource status would map to task status or not. I think it makes sense that containers depend on resource status, but I'm not sure map like this for containerStatus is necessary or not. We can remove this for now and add one for containerStatus if needed.